src/plotting/surrogate_comparison.jl

```diff
@@ -0,0 +1,13 @@
+"""
+    surrocompare(x, A, params; kwargs...) → fig
+
+Plot the surrogates of a timeseries `x` using the algorithms in `A`, with parameters in `params`.
```

Comment:
I don't undesrstand what `params` is here. Like, what is its type? 

I am assuming `A` is a tuple of `Surrogate` instances? If so, say it precisely, and rename `A` to `Surrogates`.

Reply:
I've updated the docstring as below, does it look clearer?

"""
    surrocompare(x, surrogates, params; kwargs...) → fig

Plot the surrogates of a timeseries `x` using the Cartesian product of:
- the algorithms in the iterable collection `surrogates` (`eltype(A) <: Surrogate`),
- a list of parameters passed to each algorithm, in `params`.
The elements of `params` may be numbers or collections of numbers.

## Keyword arguments
- `color`: Colors surrogate time series.
- `linewidth`: Width of the surrogate time series.
- `kwargs...`: Propagated to `Makie.Figure`.
- `resolution`: A tuple giving the resolution of the figure.
- `transient`: Number of samples to discard from the beginning of each surrogate
"""

Reply:
@brendanjohnharris  This function wouldn't work with surrogates of different type as each one would need different parameter combinations. So the current API of the function is more limitigng than it is useful. If you want a `surrocompare`, function, then make the user explicitly pass in _surrogate instances_, not just surrogate types. But in your case, I think it is simpler to just write the two loops excplicitly in the example block instead of making a new function for it.

Reply:
Also, it doesn't make much sense to compare surrogate methods on a grid like this if they don't have the same number of (conceptually related) parameters. I agree, re-usability should be paramount, moved it to the documentation.